src/main/java/org/jenkinsci/plugins/kubernetes/auth/KubernetesAuthConfig.java

```diff
@@ -18,6 +20,9 @@ public class KubernetesAuthConfig {
     private final boolean skipTlsVerify;
 
     public KubernetesAuthConfig(String serverUrl, String caCertificate, boolean skipTlsVerify) {
+        if (FIPS140.useCompliantAlgorithms() && skipTlsVerify && serverUrl.startsWith("https://")) {
```

Comment:
I guess the "only if credentials are being used" can be applied to skipping TLS verification too?

Reply:
This is to be on the safe side. Because skipping TLS verification may imply ignoring some of the FIPS recommendations (non-compliant algorithm, non-compliant certificate, etc.)

Reply:
but forcing TLS doesn't make sure you are not using a non-compliant algorithm, right? my understanding is this is why we want TLS only if there are credentials, right?

Reply:
> but forcing TLS doesn't make sure you are not using a non-compliant algorithm, right? my understanding is this is why we want TLS only if there are credentials, right?

Yes to both questions. But if this comment applies to the code above, note that the purpose of it is not to force TLS. The idea is: if FIPS is enabled and the scheme is `https`, then the TLS errors should not be ignored (`skipTlsVerify` should be false).

Sorry if my previous comment was misleading.